api/config_entry_ingress.go

```diff
@@ -0,0 +1,41 @@
+package api
+
+// IngressGatewayConfigEntry manages the configuration for an ingress service
+// with the given name.
+type IngressGatewayConfigEntry struct {
+	Kind string
+	Name string
+
+	Listeners []IngressListener
+
+	CreateIndex uint64
+	ModifyIndex uint64
+}
+
+type IngressListener struct {
```

Comment:
Since the api module is public-facing, it would be good if these types are documented for godoc.

Reply:
Also, what if we rename this file to `api/config_entry_gateways.go` and use it for both kinds?

Reply:
Makes sense to me. Done!

Reply:
Updated the `agent/structs/config_entry_ingress*` to match as well.